### DISCUSSION TITLE
Behaviour of resource.write after steps.table_normalize

### DISCUSSION THREAD

**Discussion by fjuniorr** [frictionlessdata/frictionlessdata.io#822] [Q&A]
I'm trying to use frictionless transform to standardize data sources that use different representations for the same data (think dates in different formats).

For example, for [the csv file](https://github.com/splor-mg/datapackage-reprex/blob/main/20230711T100325/input.csv) described by a [proper table schema](https://github.com/splor-mg/datapackage-reprex/blob/main/20230711T100325/resource.yaml):

![20230711T105401](https://github.com/frictionlessdata/frictionlessdata.io/assets/4969445/ce529d7c-462f-4814-ab84-e35e6acb0fa0)

I would like to generate the [following csv file](https://github.com/splor-mg/datapackage-reprex/blob/main/20230711T100325/output-petl.csv):

![20230711T110147](https://github.com/frictionlessdata/frictionlessdata.io/assets/4969445/75642854-398b-41f7-85d8-6835aca58f12)

I can achieve this by exporting the csv with `petl`:

```python
from frictionless import Resource, transform, steps

url = 'https://raw.githubusercontent.com/splor-mg/datapackage-reprex/main/20230711T100325/resource.yaml'
resource = Resource(url)
transform(resource, steps=[steps.table_normalize()])
resource.to_petl().tocsv('output-petl.csv')
```

but I'm wondering if I shoudn't be able to do the same with `resource.write`

```python
from frictionless import Resource, transform, steps

url = 'https://raw.githubusercontent.com/splor-mg/datapackage-reprex/main/20230711T100325/resource.yaml'
resource = Resource(url)
transform(resource, steps=[steps.table_normalize()])
resource.write('output-frictionless.csv')
```

which currently generates the [following csv file](https://github.com/splor-mg/datapackage-reprex/blob/main/20230711T100325/output-frictionless.csv):

![20230711T111910](https://github.com/frictionlessdata/frictionlessdata.io/assets/4969445/5b89caff-1248-4fe7-babb-fcb6449b9314)

Note that `resource.write` did converted the `missingValue` from `NA` to a empty char in the `memo` field without adjusting the tableschema.

**Comment by fjuniorr** [comment]
Just to note that 

```python
transform(resource, steps=[steps.table_normalize(), steps.table_write(path='table_write.csv')])
```

also does not output the normalized data.

**Comment by roll** [comment]
Thanks! I created an issue to investigate https://github.com/frictionlessdata/frictionless-py/issues/1567

**Comment by fjuniorr** [comment]
Hi @roll! Could you expand a little more your thinking into way this should be the behaviour of `resource.write`?

For me, if the in-memory representation of the data has changed from

```python
[['id', 'valid', 'amount', 'memo', 'valid_from', 'valid_to', 'updated_at'],
 ['1',
  'no',
  '120',
  'some information',
  '01/02/1995',
  '01/02/1995',
  '30/11/1994 09:45:00'],
 ['2',
  'yes',
  'R$ 2.000,5',
  'NA',
  '01/02/2002',
  '01/02/2002',
  '30/09/2002 10:45:00']]
```

to (after the call to `table_normalize`)

```python
[['id', 'valid', 'amount', 'memo', 'valid_from', 'valid_to', 'updated_at'],
 [1,
  False,
  Decimal('120'),
  'some information',
  datetime.date(1995, 1, 2),
  datetime.date(1995, 2, 1),
  datetime.datetime(1994, 11, 30, 9, 45)],
 [2,
  True,
  Decimal('2000.5'),
  None,
  datetime.date(2002, 1, 2),
  datetime.date(2002, 2, 1),
  datetime.datetime(2002, 9, 30, 10, 45)]]
```

why would `resource.write` not export `Decimal` and `datetime` as a standard csv python export:

```python
data = resource.read_cells()
import csv
with open('data.csv', 'w') as csvfile:
    # creating a csv writer object
    csvwriter = csv.writer(csvfile)
    csvwriter.writerow(data[0])
    csvwriter.writerows(data[1:])
```

**Reply by roll** [reply]
@fjuniorr 
It uses the source schema, so we need to provide a target schema to change the format (steps.table_write(path=path, schema=new_schema))

**Reply by fjuniorr** [reply]
I don't think I follow @roll. [`steps.table_write`](https://github.com/frictionlessdata/frictionless-py/blob/main/frictionless/steps/table/table_write.py) does't look like it takes a `schema` argument and indeed I get:

```python
from frictionless import Resource, transform, steps, Schema

resource = Resource('https://raw.githubusercontent.com/splor-mg/datapackage-reprex/main/reprex/20230711T100325/resource.yaml')
target_schema = Schema('https://raw.githubusercontent.com/splor-mg/datapackage-reprex/main/reprex/20230711T100325/target.yaml')

transform(resource, steps=[steps.table_normalize(), steps.table_write(path='output.csv', schema=target_schema)])
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: table_write.__init__() got an unexpected keyword argument 'schema'
```

with `frictionless` 5.15.10.

**Reply by roll** [reply]
@shashigharti Can you please take a look?
